### PR TITLE
Fix Video Recoding

### LINKED ISSUE #173: Recording a Movie let Webots Crashes
**Describe the Bug**
Webots crashes when recording a movie.

**Steps to Reproduce**
1. Record any movie.

**System**
 - Operating System: Windows 10
 - Graphics Card: AMD Radeon R5

### PR DESCRIPTION
**Pull request by DavidMansolino**
Fix #173.

(Quick and dirty patch for testing for now).

### CODE REVIEW

**Comment by DavidMansolino** [comment]
test me

**Comment by DavidMansolino** [comment]
👍

File: src/webots/gui/WbVideoRecorder.cpp [hunk 1]
```diff
@@ -56,22 +56,26 @@ class FrameWriterThread : public QThread {
...
+    const int w = mResolution.width() / mPixelRatio;
+    const int h = mResolution.height() / mPixelRatio;
+    mFrame = new unsigned char[4 * w * h];
+    WbView3D::flipAndScaleDownImageBuffer(frame, mFrame, mResolution.width(), mResolution.height(), mPixelRatio);
+  }
+
+  virtual ~FrameWriterThread() { delete[] mFrame; }
 
   void run() override {
     mView->lockPBOMutex(mMutexIndex);
```
Comment (stefaniapedrazzi) [inline src/webots/gui/WbVideoRecorder.cpp]: Is this still required now that the frame is copied and no longer read directly from the PBO?

Comment (DavidMansolino) [reply src/webots/gui/WbVideoRecorder.cpp]: Hum.. good point, probably we can remove it, let me check it.

**Comment by DavidMansolino** [comment]
test me

**Comment by DavidMansolino** [comment]
👍

File: src/webots/gui/WbVideoRecorder.cpp [hunk 2]
```diff
@@ -56,22 +56,25 @@ class FrameWriterThread : public QThread {
 public:
   FrameWriterThread(unsigned char *frame, int mutexIndex, const QString &fileName, const QSize &resolution, int pixelRatio,
                     int quality, WbView3D *view) :
-    mFrame(frame),
     mMutexIndex(mutexIndex),
```
Comment (stefaniapedrazzi) [inline src/webots/gui/WbVideoRecorder.cpp]: This class member is also no longer needed.

**Comment by stefaniapedrazzi** [comment]
I did some quick speed test based on the speedometer and I didn't noticeable difference on my Linux system while recording a video at resolution 1280x720 (in fullscreen mode I don't see the speedometer).

**Comment by DavidMansolino** [comment]
Thank you for the test, I checked on Windows on two different systems and didn't notice any speed difference either.

**Comment by DavidMansolino** [comment]
test me

**Comment by DavidMansolino** [comment]
test me

**Comment by DavidMansolino** [comment]
:+1:

**Pull request merged**